packages/terra-pills/src/private/_Pill.jsx

```diff
@@ -0,0 +1,313 @@
+      if (isTruncated) {
+        setPopupOpen(true);
+      }
+    } else if ((event.keyCode === KEY_DELETE || event.keyCode === KEY_BACK_SPACE)) {
+      event.preventDefault();
+      if (onRemove) {
+        onRemove(pillKey, metaData, event);
+      }
+    }
+  };
+
+  const handleOnMouseDown = () => {
+    pillRef.current.setAttribute('data-terra-pills-show-focus-styles', 'false');
+  };
+
+  const handleOnBlur = () => {
+    pillRef.current.setAttribute('data-terra-pills-show-focus-styles', 'true');
+  };
+
+  const getPillRef = () => pillRef.current;
+
+  const pillInteraction = {};
+  pillInteraction.isSelectable = isTruncated;
+  pillInteraction.isRemovable = onRemove;
+  pillInteraction.isSelectableAndRemovable = (onRemove && isTruncated);
+
+  const pillProps = {};
+  pillProps.title = title || (isTruncated ? label : undefined);
+  if (pillInteraction.isSelectable || pillInteraction.isRemovable) {
+    pillProps.tabIndex = '0';
+    pillProps.onKeyDown = handleOnKeyDown;
+    pillProps.onMouseDown = handleOnMouseDown;
+    pillProps.onBlur = handleOnBlur;
+  }
+
+  const pillButtonProps = {};
+  if (pillInteraction.isSelectable) {
+    pillButtonProps.onClick = handleOnClick;
+  }
+
+  const removeButtonProps = {};
+  if (onRemove) {
+    removeButtonProps.onClick = handleOnRemove;
+  }
+
+  let pillInteractionHint;
+  if (pillInteraction.isSelectableAndRemovable) {
+    pillInteractionHint = `, ${intl.formatMessage({ id: 'Terra.pills.hint.selectableAndRemovable' })}`;
+  } else if (pillInteraction.isSelectable) {
+    pillInteractionHint = `, ${intl.formatMessage({ id: 'Terra.pills.hint.selectable' })}`;
+  } else if (pillInteraction.isRemovable) {
+    pillInteractionHint = `, ${intl.formatMessage({ id: 'Terra.pills.hint.removable' })}`;
+  }
+
+  const pillClassNames = classNames(
+    cx([
+      'pill',
+      { 'is-focusable': pillInteraction.isSelectable || pillInteraction.isRemovable },
+      { 'is-selectable': pillInteraction.isSelectable },
+      { 'is-removable': pillInteraction.isRemovable },
+      { 'is-selectable-and-removable': pillInteraction.isSelectableAndRemovable },
+      theme.className,
+    ]),
+    customProps.className,
+  );
+
+  const renderTruncatedLabelPopup = () => (
+    <Popup
+      isOpen={open}
+      isArrowDisplayed
+      targetRef={getPillRef}
+      onRequestClose={handleOnRequestClose}
+      contentHeight="auto"
+      contentWidth="auto"
+    >
+      <div className={cx(['popup-content-pill-label', theme.className])}>
+        {labelCategory
+          ? <span className={cx(['popup-content-pill-category', theme.className])}>{`${labelCategory}: `}</span>
+          : undefined}
+        <span>{label}</span>
+      </div>
+    </Popup>
+  );
+
+  const renderRemoveButton = () => (
+    <button
+      {...removeButtonProps}
+      className={cx('pill-remove-button')}
+      tabIndex="-1"
+      type="button"
+      aria-labelledby={`remove-button-${id}`}
+      aria-hidden="true"
```

Comment:
Why does this button have `aria-hidden` set to true?

Reply:
aria-hidden is removed.